substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/dcmd/JfrDumpDcmd.java

```diff
@@ -0,0 +1,314 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.jfr.dcmd;
+
+import com.oracle.svm.core.dcmd.AbstractDcmd;
+import com.oracle.svm.core.dcmd.DcmdOption;
+import com.oracle.svm.core.dcmd.DcmdParseException;
+import com.oracle.svm.core.jfr.JfrArgumentParser.JfrArgument;
+import com.oracle.svm.core.jfr.JfrArgumentParser.JfrArgumentParsingFailed;
+import com.oracle.svm.core.jfr.JfrManager;
+import com.oracle.svm.core.util.BasedOnJDKFile;
+
+import jdk.jfr.FlightRecorder;
+import jdk.jfr.Recording;
+import jdk.jfr.internal.PlatformRecorder;
+import jdk.jfr.internal.PlatformRecording;
+import jdk.jfr.internal.SecuritySupport;
+import jdk.jfr.internal.WriteableUserPath;
+import jdk.jfr.internal.PrivateAccess;
+
+import java.io.IOException;
+import java.nio.file.InvalidPathException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Map;
+
+import static com.oracle.svm.core.jfr.JfrArgumentParser.DumpArgument;
+import static com.oracle.svm.core.jfr.JfrArgumentParser.parseBoolean;
+import static com.oracle.svm.core.jfr.JfrArgumentParser.parseDuration;
+import static com.oracle.svm.core.jfr.JfrArgumentParser.parseJfrOptions;
+import static com.oracle.svm.core.jfr.JfrArgumentParser.parseMaxSize;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class JfrDumpDcmd extends AbstractDcmd {
```

Comment:
Why can't we use `jdk.jfr.internal.dcmd.DCmdDump` instead?

Reply:
Maybe we could. But `jdk.jfr.internal.dcmd.DCmdDump` expects to be provided with a `jdk.jfr.internal.dcmd.ArgumentParser` and also writes back results differently (`reportOperationComplete`). In OpenJDK, each individual DCMD is responsible for sending back their own specific responses.  In this PR, only the AttachListener classes know about the sockets and the DCMDs just return a simple string for the AttachListener to deal with.